src/presign/record.rs

```diff
@@ -262,30 +262,110 @@ mod tests {
         elliptic_curve::{Field, Group},
         ProjectivePoint, Scalar,
     };
-    use rand::{rngs::StdRng, Rng, SeedableRng};
+    use libpaillier::unknown_order::BigNumber;
+    use rand::{rngs::StdRng, CryptoRng, Rng, RngCore, SeedableRng};
 
     use crate::{
-        presign::record::RECORD_TAG,
-        utils::{testing::init_testing, CurvePoint},
-        PresignRecord,
+        keygen,
+        presign::{participant::presign_record_set_is_valid, record::RECORD_TAG},
+        utils::{bn_to_scalar, testing::init_testing, CurvePoint},
+        ParticipantConfig, PresignRecord,
     };
 
-    fn random_record(rng: &mut StdRng) -> PresignRecord {
-        let point = CurvePoint(ProjectivePoint::random(StdRng::from_seed(rng.gen())));
-        let random_share = Scalar::random(StdRng::from_seed(rng.gen()));
-        let chi_share = Scalar::random(rng);
+    impl PresignRecord {
+        pub(crate) fn mask_point(&self) -> &CurvePoint {
+            &self.R
+        }
+
+        pub(crate) fn mask_share(&self) -> &Scalar {
+            &self.k
+        }
 
-        PresignRecord {
-            R: point,
-            k: random_share,
-            chi: chi_share,
+        pub(crate) fn masked_key_share(&self) -> &Scalar {
+            &self.chi
```

Comment:
I put these utility methods in `mod test` because normally we don't have to access presign record fields directly, but they were helpful to have for validating the record in `presign::participant.rs`.